src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -662,8 +663,13 @@ private static void EnsureInitializedCore()
             {
                 if (!s_initialized)
                 {
-                    // Ensure the console is configured appropriately.  This will start
-                    // signal handlers, etc.
+                    // Ensure signal handling is setup.
+                    if (!Interop.Sys.InitializeSignalHandling())
+                    {
+                        throw new Win32Exception();
+                    }
+
+                    // Ensure the console is configured appropriately.
                     if (!Interop.Sys.InitializeConsole())
```

Comment:
Should InitializeConsole just call InitializeSignalHandling rather than making two interop calls?